### PR TITLE
PMM-5706 Installed Grafana plugins are removed after upgrade

### PR DESCRIPTION
**Pull request by adivinho**
- [ ] https://github.com/Percona-Lab/pmm-submodules/pull/883

### CODE REVIEW

File: import-dashboards.py [hunk 1]
```diff
@@ -231,17 +231,17 @@ def add_panels():
         if os.path.isdir(source_dir):
```
Comment (askomorokhov) [inline import-dashboards.py:231]: While we change this function, a few things:
`for app in ['panels']: ` - this loop looks like rudimental. Can we remove it?


 ```
source_dir = '/usr/share/percona-dashboards/' + app
dest_dir = '/var/lib/grafana/plugins/'
```
^^^ - according to these lines. Can we name these paths as constants (in upper case.)?

File: import-dashboards.py [hunk 2]
```diff
@@ -231,17 +231,17 @@ def add_panels():
...
-            try:
-                shutil.rmtree(dest_dir, False)
-            except Exception as err:
-                print '  * Failed to remove %s: %s' % (dest_dir, err)
-            shutil.copytree(source_dir, dest_dir)
+            if not os.path.isdir(dest_dir):
+                shutil.copytree(source_dir, dest_dir)
+            else:
+                for file in files_list:
+                    shutil.copyfile(os.path.join(source_dir,file), os.path.join(dest_dir,file))
```
Comment (askomorokhov) [inline import-dashboards.py:238]:
```suggestion
                    shutil.copyfile(os.path.join(source_dir, file), os.path.join(dest_dir, file))
```

File: import-dashboards.py [hunk 3]
```diff
@@ -231,17 +231,17 @@ def add_panels():
...
             print '  * Unzipping %r' % (app,)
             for file in files_list:
-                with zipfile.ZipFile(dest_dir + file, 'r') as zip_ref:
+                with zipfile.ZipFile(os.path.join(dest_dir,file), 'r') as zip_ref:
```
Comment (askomorokhov) [inline import-dashboards.py:241]:
```suggestion
                with zipfile.ZipFile(os.path.join(dest_dir, file), 'r') as zip_ref:
```

File: import-dashboards.py [hunk 4]
```diff
@@ -231,17 +231,17 @@ def add_panels():
...
                     print '   * Unzip %r' % (file,)
                     zip_ref.extractall(dest_dir)
-                os.remove(dest_dir + file)
+                os.remove(os.path.join(dest_dir,file))
```
Comment (askomorokhov) [inline import-dashboards.py:244]:
```suggestion
                os.remove(os.path.join(dest_dir, file))
```

File: import-dashboards.py [hunk 5]
```diff
@@ -255,8 +255,19 @@ def rename_panels():
             if os.path.exists(panel_path):
```
Comment (askomorokhov) [inline import-dashboards.py:255]: `for app in ['panels']: `  can we remove this loop?

File: import-dashboards.py [hunk 6]
```diff
@@ -255,8 +255,19 @@ def rename_panels():
...
                 with open(panel_path, 'r') as f:
                     panel_params = json.loads(f.read())
+                    if panel == panel_params['id']:
```
Comment (askomorokhov) [inline import-dashboards.py:258]: Could you please add a comment on why we need to skip in this case?

File: import-dashboards.py [hunk 7]
```diff
@@ -255,8 +255,19 @@ def rename_panels():
...
                     print '%r' % (panel_params['id'],)
+                    if os.path.isdir(os.path.join(GRAFANA_PLUGINS_DR, panel_params['id'])):
+                        try:
+                            shutil.rmtree(os.path.join(GRAFANA_PLUGINS_DR, panel_params['id']))
+                        except Exception as err:
+                            print '  * Failed to remove %s: %s' % (os.path.join(GRAFANA_PLUGINS_DR, panel_params['id']), err)
+                            continue
                     os.rename(os.path.join(GRAFANA_PLUGINS_DR, panel), os.path.join(GRAFANA_PLUGINS_DR, panel_params['id']))
+            else:
+                print 'skipped'
```
Comment (askomorokhov) [inline import-dashboards.py:270]:
```suggestion
                print('Path %r does not exist - skipped' % (panel_path,))
```
Too far from `if` - maybe this msg or please add a comment.

Comment (adivinho) [reply import-dashboards.py]: I have added more details into the output message.

File: import-dashboards.py (hunk 6 above)
Comment (adivinho) [reply import-dashboards.py]: I have added more details into the output message. So it has the reason of using the checking.

File: import-dashboards.py (hunk 1 above)
Comment (adivinho) [reply import-dashboards.py]: It has been transformed into a constant.

File: import-dashboards.py [hunk 8]
```diff
@@ -224,39 +225,47 @@ def add_datasources(api_key):
...
-                print '  * Failed to remove %s: %s' % (dest_dir, err)
-            shutil.copytree(source_dir, dest_dir)
-            print '  * Unzipping %r' % (app,)
+    print ' * Adding panels'
+    if os.path.isdir(GRAFANA_SOURCE_PLUGINS_DR):
+        files_list = os.listdir(GRAFANA_SOURCE_PLUGINS_DR)
+        print '  * Copying panels'
+        if not os.path.isdir(GRAFANA_PLUGINS_DR):
+            shutil.copytree(GRAFANA_SOURCE_PLUGINS_DR, GRAFANA_PLUGINS_DR)
+        else:
```
Comment (percona-csalguero) [inline import-dashboards.py:234]: According to copytree doc https://docs.python.org/3/library/shutil.html#shutil.copytree , there is a dirs_exist_ok parameter we can use and keep only the shutil.copytree part and remove the else and the for loop.
Is that correct or there is another reason to have the for loop?

File: import-dashboards.py [hunk 9]
```diff
@@ -23,19 +23,20 @@
...
-NEW_VERSION_FILE   = SCRIPT_DIR + '/VERSION'
-OLD_VERSION_FILE   = GRAFANA_DB_DIR + '/PERCONA_DASHBOARDS_VERSION'
-HOST               = 'http://127.0.0.1:3000'
-LOGO_FILE          = '/usr/share/pmm-server/landing-page/img/pmm-logo.svg'
-SET_OF_TAGS        = {'Query Analytics': 0, 'OS': 0, 'MySQL': 0, 'MongoDB': 0, 'PostgreSQL': 0, 'Insight': 0, 'PMM': 0}
-YEAR               = str(datetime.date.today())[:4]
-
-CONTENT            = '''<center>
+GRAFANA_DB_DIR            = sys.argv[1] if len(sys.argv) > 1 else '/srv/grafana'
+GRAFANA_IMG_DR            = '/usr/share/grafana/public/img/'
```
Comment (percona-csalguero) [inline import-dashboards.py:27]: This is not mandatory, but if we can, we should use DR or DIR everywhere

**Comment by adivinho** [comment]
Yes. It has already tested.

**Pull request merged**